repository.go

```diff
@@ -453,3 +454,129 @@ func (r *Repository) Reference(name plumbing.ReferenceName, resolved bool) (
 func (r *Repository) References() (storer.ReferenceIter, error) {
 	return r.s.IterReferences()
 }
+
+// ResolveRevision resolves revision to corresponding hash
+func (r *Repository) ResolveRevision(rev plumbing.Revision) (*plumbing.Hash, error) {
+	p := revision.NewParserFromString(string(rev))
```

Comment:
Also the fact that a revision (`plumbing.Revision`) is not part of the revision package is very confusing:

The revision package does not define a revision (confusing), it defines a parser for a revision instead (confusing).

Nobody should care about revision parsers, as they should be private code, the revision should have methods resolve itself without forcing you to use its internal parser.

So either the name of the package is wrong (revision should be revisionParser) or things have to be refactored to make sense.